benchalerts/clients.py

```diff
@@ -0,0 +1,209 @@
+        if not adapter:
+            retry_strategy = Retry(
+                total=5,
+                status_forcelist=frozenset((429, 502, 503, 504)),
+                backoff_factor=4,  # will retry in 2, 4, 8, 16, 32 seconds
+            )
+            adapter = HTTPAdapter(max_retries=retry_strategy)
+
+        self.session = requests.Session()
+        self.session.mount("https://", adapter)
+
+    def get(self, path: str) -> dict:
+        url = self.base_url + path
+        log.debug(f"GET {url}")
+        res = self.session.get(url=url, timeout=self.timeout_s)
+        res.raise_for_status()
+        return res.json()
+
+    def post(self, path: str, json: dict) -> Optional[dict]:
+        url = self.base_url + path
+        log.debug(f"POST {url} {json}")
+        res = self.session.post(url=url, json=json, timeout=self.timeout_s)
+        res.raise_for_status()
+        if res.content:
+            return res.json()
+
+
+class GithubRepoClient(_BaseClient):
+    """A client to interact with a Github repo.
+
+    Parameters
+    ----------
+    repo
+        The repo name, in the form 'owner/repo'.
+    adapter
+        A requests adapter to mount to the requests session. If not given, one will be
+        created with a backoff retry strategy.
+
+    Environment variables
+    ---------------------
+    GITHUB_API_TOKEN
+        A Github API token with ``repo`` access.
+    """
+
+    def __init__(self, repo: str, adapter: Optional[HTTPAdapter] = None):
+        token = os.getenv("GITHUB_API_TOKEN")
+        if not token:
+            fatal_and_log("Environment variable GITHUB_API_TOKEN not found")
+
+        super().__init__(adapter=adapter)
+        self.session.headers = {"Authorization": f"Bearer {token}"}
+        self.base_url = f"https://api.github.com/repos/{repo}"
+
+    def create_pull_request_comment(
+        self,
+        comment: str,
+        *,
+        pull_number: Optional[int] = None,
+        commit_sha: Optional[str] = None,
+    ):
+        """Create a comment on a pull request, specified either by pull request number
+        or commit SHA.
+
+        Parameters
+        ----------
+        comment
+            The comment text.
+        pull_number
+            The number of the pull request. Specify either this or ``commit_sha``.
+        commit_sha
+            The SHA of a commit associated with the pull request. Specify either this
+            or ``pull_number``.
+        """
+        if not pull_number and not commit_sha:
+            fatal_and_log("pull_number and commit_sha are both missing")
+
+        if commit_sha:
+            pull_numbers = [
+                pull["number"] for pull in self.get(f"/commits/{commit_sha}/pulls")
+            ]
+            if len(pull_numbers) != 1:
+                fatal_and_log(
+                    "Need exactly 1 pull request associated with commit "
+                    f"'{commit_sha}'. Found {pull_numbers}."
+                )
```

Comment:
Not quite sure how a commit would end up associated with multiple PRs (though I'm sure it's possible somehow), but this seems like exactly the situation where somebody might want to specify both `pull_number` and `commit_sha` to disambiguate`. An infrequent corner-case, though; not worth lots of effort

Reply:
Yeah I think I'll ignore this corner-case for now and come back to it if there's a problem. 👍